affine/chutes.py

```diff
@@ -60,6 +60,42 @@ async def check_model_gated(model_id: str, revision: Optional[str] = None) -> Op
             return cached[0]
         return None
 
+# --------------------------------------------------------------------------- #
+#                               HF weights SHA                                #
+# --------------------------------------------------------------------------- #
+async def get_weights_sha(model_id: str, revision: Optional[str] = None) -> Optional[str]:
+    try:
+        def _repo_info():
+            api = HfApi(token=os.getenv("HF_TOKEN", None))
+            return api.repo_info(repo_id=model_id, repo_type="model", revision=revision, files_metadata=True)
+        info = await asyncio.to_thread(_repo_info)
+
+        siblings = getattr(info, "siblings", None) or []
+
+        def _name(s) -> str:
+            return (getattr(s, "rfilename", None) or getattr(s, "path", "") or "")
+
+        # 1) Prefer exact 'model.safetensors'
+        for s in siblings:
+            name = _name(s)
+            lfsi = getattr(s, "lfs", None) or {}
+            sha = lfsi.get("sha256") if isinstance(lfsi, dict) else None
+            if sha and name.endswith("model.safetensors"):
+                return str(sha)
+
+        # 2) Any '*.safetensors'
+        for s in siblings:
+            name = _name(s)
+            lfsi = getattr(s, "lfs", None) or {}
+            sha = lfsi.get("sha256") if isinstance(lfsi, dict) else None
+            if sha and name.endswith(".safetensors"):
```

Comment:
IMO `sha512(''.join([shard.hash for shard in siblings]))` would be better than picking only one file - what if I train only the input embeddings? or put my LoRA into a separate file?